# URL
https://catalog.data.gov/dataset/behavioral-risk-factor-surveillance-system-brfss-national-cardiovascular-disease-surveilla

In [1]:
import pandas as pd

# Assuming the file path is '/mnt/data/Behavioral_Risk_Factor_Surveillance_System__BRFSS__-__National_Cardiovascular_Disease_Surveillance_Data.csv'
file_path = 'Behavioral_Risk_Factor_Surveillance_System__BRFSS__-__National_Cardiovascular_Disease_Surveillance_Data.csv'

# Reading the CSV file into a DataFrame
heart_disease_data = pd.read_csv(file_path)

# Display the first few rows of the dataframe
print(heart_disease_data.head())

                                        RowId  YearStart LocationAbbr  \
0  BRFSS~2011~01~BR001~OVR01~Age-Standardized       2011           AL   
1             BRFSS~2011~01~BR001~OVR01~Crude       2011           AL   
2             BRFSS~2011~01~BR001~GEN01~Crude       2011           AL   
3  BRFSS~2011~01~BR001~GEN01~Age-Standardized       2011           AL   
4  BRFSS~2011~01~BR001~GEN02~Age-Standardized       2011           AL   

  LocationDesc DataSource PriorityArea1  PriorityArea2 PriorityArea3  \
0      Alabama      BRFSS           NaN            NaN           NaN   
1      Alabama      BRFSS           NaN            NaN           NaN   
2      Alabama      BRFSS           NaN            NaN           NaN   
3      Alabama      BRFSS           NaN            NaN           NaN   
4      Alabama      BRFSS           NaN            NaN           NaN   

   PriorityArea4                    Class  ... Break_Out_Category Break_Out  \
0            NaN  Cardiovascular Diseases  ...   

C:\Users\bmcki\AppData\Local\Temp\ipykernel_30160\4009407940.py:7: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  heart_disease_data = pd.read_csv(file_path)


In [2]:
# Check the rows and columns
heart_disease_data.shape

(160160, 30)

In [3]:
heart_disease_data.columns

Index(['RowId', 'YearStart', 'LocationAbbr', 'LocationDesc', 'DataSource',
       'PriorityArea1', 'PriorityArea2', 'PriorityArea3', 'PriorityArea4',
       'Class', 'Topic', 'Question', 'Data_Value_Type', 'Data_Value_Unit',
       'Data_Value', 'Data_Value_Alt', 'Data_Value_Footnote_Symbol',
       'Data_Value_Footnote', 'Low_Confidence_Limit', 'High_Confidence_Limit',
       'Break_Out_Category', 'Break_Out', 'ClassId', 'TopicId', 'QuestionId',
       'Data_Value_TypeID', 'BreakOutCategoryId', 'BreakOutId', 'LocationId',
       'Geolocation'],
      dtype='object')

In [4]:
heart_disease_data.isna().sum()


RowId                              0
YearStart                          0
LocationAbbr                       0
LocationDesc                       0
DataSource                         0
PriorityArea1                 137280
PriorityArea2                 160160
PriorityArea3                  80080
PriorityArea4                 160160
Class                              0
Topic                              0
Question                           0
Data_Value_Type                    0
Data_Value_Unit                    0
Data_Value                     81923
Data_Value_Alt                     0
Data_Value_Footnote_Symbol     78237
Data_Value_Footnote            78237
Low_Confidence_Limit           83847
High_Confidence_Limit          83847
Break_Out_Category                 0
Break_Out                          0
ClassId                            0
TopicId                            0
QuestionId                         0
Data_Value_TypeID                  0
BreakOutCategoryId                 0
B

In [5]:
# Drop columns with any missing values based on the provided summary
columns_to_drop_due_to_missing = [
    'PriorityArea1', 'PriorityArea2', 'PriorityArea3', 'PriorityArea4',
    'Data_Value', 'Data_Value_Footnote_Symbol', 'Data_Value_Footnote',
    'Low_Confidence_Limit', 'High_Confidence_Limit', 'Geolocation'
]

heart_disease_data = heart_disease_data.drop(columns=columns_to_drop_due_to_missing)

# Display summary to verify changes
heart_disease_data.isna().sum()


RowId                 0
YearStart             0
LocationAbbr          0
LocationDesc          0
DataSource            0
Class                 0
Topic                 0
Question              0
Data_Value_Type       0
Data_Value_Unit       0
Data_Value_Alt        0
Break_Out_Category    0
Break_Out             0
ClassId               0
TopicId               0
QuestionId            0
Data_Value_TypeID     0
BreakOutCategoryId    0
BreakOutId            0
LocationId            0
dtype: int64

In [6]:
# Check updated rows and columns to ensure no loss data
heart_disease_data.shape

(160160, 20)

In [7]:
# Check existing columns
heart_disease_data.columns

Index(['RowId', 'YearStart', 'LocationAbbr', 'LocationDesc', 'DataSource',
       'Class', 'Topic', 'Question', 'Data_Value_Type', 'Data_Value_Unit',
       'Data_Value_Alt', 'Break_Out_Category', 'Break_Out', 'ClassId',
       'TopicId', 'QuestionId', 'Data_Value_TypeID', 'BreakOutCategoryId',
       'BreakOutId', 'LocationId'],
      dtype='object')

In [8]:
# Check for unique ids
heart_disease_data['RowId'].nunique()

160160

In [9]:
# Number of unique ids
unique_ids_count = heart_disease_data['RowId'].nunique()
f"There are {unique_ids_count} unique IDs in the dataset."

'There are 160160 unique IDs in the dataset.'

In [10]:
# Years of frequency
year_frequency = heart_disease_data['YearStart'].value_counts().sort_values(ascending=False)
year_frequency

YearStart
2011    16016
2012    16016
2013    16016
2014    16016
2015    16016
2016    16016
2017    16016
2018    16016
2019    16016
2020    16016
Name: count, dtype: int64

## There are 10 years with 16016 entries which makes the dataset consistant.

In [11]:
# Locations abbreviations
loc_abbr = heart_disease_data['LocationAbbr'].value_counts().sort_values(ascending=False)
loc_abbr

LocationAbbr
AL     3080
AK     3080
ID     3080
AZ     3080
AR     3080
CA     3080
CO     3080
CT     3080
DE     3080
DC     3080
FL     3080
GA     3080
HI     3080
IL     3080
MS     3080
IN     3080
IA     3080
KS     3080
KY     3080
LA     3080
ME     3080
MD     3080
MA     3080
MI     3080
MN     3080
MO     3080
MT     3080
NE     3080
PA     3080
NV     3080
NH     3080
NJ     3080
NM     3080
NY     3080
NC     3080
ND     3080
OH     3080
OK     3080
OR     3080
RI     3080
WY     3080
SC     3080
SD     3080
TN     3080
TX     3080
UT     3080
VT     3080
VA     3080
WA     3080
WV     3080
WI     3080
USM    3080
Name: count, dtype: int64

In [12]:
# Unique values for 'LocationAbbr' column
unique_loc_abbr_count = heart_disease_data['LocationAbbr'].nunique()
unique_loc_abbr_count

52

In [13]:
# Assuming each location abbreviation (LocationAbbr) has 3080 entries
# and that there are 52 unique location abbreviations

# Number of entries per location abbreviation
entries_per_location = 3080

# Total number of unique location abbreviations
unique_locations = 52

# Total number of entries
total_entries = entries_per_location * unique_locations

total_entries

160160

## There are a total of 52 states with 3080 per location making the data consistant.

In [14]:
# Unique values for DataSource column
heart_disease_data['DataSource'].nunique()

1

# Only one unique datasource: BRFSS

In [15]:
# Unique values for Class column
heart_disease_data['Class'].value_counts()

Class
Risk Factors               102960
Cardiovascular Diseases     57200
Name: count, dtype: int64

# Risk Factors are higher which make outside influences and models to look into deeper

In [16]:
# Unique values for Topic column
heart_disease_data['Topic'].value_counts()

Topic
Acute Myocardial Infarction (Heart Attack)    22880
Cholesterol Abnormalities                     22880
Hypertension                                  22880
Major Cardiovascular Disease                  11440
Coronary Heart Disease                        11440
Stroke                                        11440
Diabetes                                      11440
Nutrition                                     11440
Obesity                                       11440
Physical Inactivity                           11440
Smoking                                       11440
Name: count, dtype: int64

## Three columns(Acute Myocardial Infarction (Heart Attack), Cholesterol Abnormalities, and Hypertension) have 22880 values at the highest, and the remaning eight values have the lowers of 11440 values. Can create different graphs with models to see the uniqueness in each. 

In [17]:
# Unique values for Question column
heart_disease_data['Question'].value_counts()

Question
Prevalence of major cardiovascular disease among US adults (18+); BRFSS                                  11440
Prevalence of acute myocardial infarction (heart attack) among US adults (18+); BRFSS                    11440
Prevalence of post-hospitalization rehabilitation among heart attack patients, US adults (18+); BRFSS    11440
Prevalence of coronary heart disease among US adults (18+); BRFSS                                        11440
Prevalence of stroke among US adults (18+); BRFSS                                                        11440
Prevalence of cholesterol screening in the past 5 years among US adults (20+); BRFSS                     11440
Prevalence of high total cholesterol among US adults (20+); BRFSS                                        11440
Prevalence of diabetes among US adults (18+); BRFSS                                                      11440
Prevalence of consuming fruits and vegetables less than 5 times per day among US adults (18+); BRFSS   

# All questions have the same values count of 11440

In [18]:
# Unique values for Data_Value_Type column
heart_disease_data['Data_Value_Type'].value_counts()

Data_Value_Type
Crude               101920
Age-Standardized     58240
Name: count, dtype: int64

## The crude death rate does not reflect changes in the age structure of the population over time. Age-standardisation is a method of adjustment to allow for the effect of variation in the population age structure when comparing death rates for different years or different locations. Great to look at different models to check the influences.

In [19]:
# Unique values for Data_Value_Unit column
heart_disease_data['Data_Value_Unit'].value_counts()

Data_Value_Unit
Percent (%)    160160
Name: count, dtype: int64

# All data values go by a percentage

In [20]:
# Unique values for Data_Value_Alt column
heart_disease_data['Data_Value_Alt'].value_counts()

Data_Value_Alt
-1.0     59500
-2.0     22423
 3.0       504
 2.9       502
 3.3       493
         ...  
 48.3        6
 99.3        5
 99.1        4
 0.1         2
 99.4        1
Name: count, Length: 996, dtype: int64

# Will check back later to check on how the correlation impacts the data

In [21]:
# Unique values for Break_Out_Category column
heart_disease_data['Break_Out_Category'].value_counts()

Break_Out_Category
Race       72800
Age        43680
Gender     29120
Overall    14560
Name: count, dtype: int64

# There are different values in each of the 4 categories. Race is impacted the most, while all three combined (Race, Age, Gender), are the lowest. Definitely a great factor to see why there are more values individually that grouped.

In [22]:
# Unique values for Break_Out column
heart_disease_data['Break_Out'].value_counts()

Break_Out
Overall               14560
Male                  14560
Female                14560
Non-Hispanic White    14560
Non-Hispanic Black    14560
Non-Hispanic Asian    14560
Hispanic              14560
Other                 14560
25-44                  7280
45-64                  7280
65+                    7280
35+                    7280
75+                    7280
18-24                  5720
20-24                  1560
Name: count, dtype: int64

# Different values for all age groups and races. Would be interesting to look at both groups individually and groups to see the impact as shown in the 'Break_Out_Category' column

## Check back on columns with IDs:
"ClassId", "TopicId", "QuestionId", "Data_Value_TypeID", "BreakOutCategoryId", "BreakOutId", and "LocationId"

In [24]:
heart_disease_data.to_excel('heart_disease_data_cleaned.xlsx', index=False)

# Export cleaned up data